src/services/lnd/onchain-service.ts

```diff
@@ -48,3 +44,19 @@ export const OnChainService = (
     getIncomingTransactions,
   }
 }
+
+export const extractIncomingTransactions = (
+  decoder: TxDecoder,
+  { transactions }: GetChainTransactionsResult,
+): SubmittedTransaction[] => {
+  return transactions
+    .filter((tx) => !tx.is_outgoing && !!tx.transaction)
```

Comment:
Can we have a descriptive name for this filtering? What dose `!tx.is_outgoing && !!tx.transaction` mean?

Can we also possibly do the filtering/mapping in one loop? Either with `reduce` or `flatMap`